Code/Framework/AzCore/Tests/Memory/AllocatorBenchmarks.cpp

```diff
@@ -0,0 +1,484 @@
+            m_allocations.shrink_to_fit();
+
+            AllocatorBenchmarkFixture<TAllocator>::internalTearDown(state);
+        }
+    public:
+        void Benchmark(benchmark::State& state)
+        {
+            for (auto _ : state)
+            {
+                state.PauseTiming();
+                const size_t processMemoryBaseline = Platform::GetProcessMemoryUsageBytes();
+
+                const size_t numberOfAllocations = m_allocations.size();
+                for (size_t allocationIndex = 0; allocationIndex < numberOfAllocations; ++allocationIndex)
+                {
+                    const AllocationSizeArray& allocationArray = s_allocationSizes[TAllocationSize];
+                    const size_t allocationSize = allocationArray[allocationIndex % allocationArray.size()];
+                    m_allocations[allocationIndex] = TestAllocatorType::Allocate(allocationSize, 0);
+                }
+
+                for (size_t allocationIndex = 0; allocationIndex < numberOfAllocations; ++allocationIndex)
+                {
+                    const AllocationSizeArray& allocationArray = s_allocationSizes[TAllocationSize];
+                    const size_t allocationSize = allocationArray[allocationIndex % allocationArray.size()];
+                    state.ResumeTiming();
+                    TestAllocatorType::DeAllocate(m_allocations[allocationIndex], allocationSize);
+                    state.PauseTiming();
+                    m_allocations[allocationIndex] = nullptr;
+                }
+
+                state.counters[s_counterAllocatorMemoryRatio] =
+                    benchmark::Counter(static_cast<double>(TestAllocatorType::NumAllocatedBytes()), benchmark::Counter::kDefaults);
+                state.counters[s_counterProcessMemoryRatio] = benchmark::Counter(
+                    static_cast<double>(Platform::GetProcessMemoryUsageBytes() - processMemoryBaseline),
+                    benchmark::Counter::kDefaults);
+
+                state.SetItemsProcessed(numberOfAllocations);
+
+                TestAllocatorType::GarbageCollect();
+
+                state.ResumeTiming();
+            }
+        }
+
+    private:
+        AZStd::vector<void*> m_allocations;
+    };
+
+    static void RunRanges(benchmark::internal::Benchmark* b)
+    {
+        for (int i = 0; i < 6; ++i)
+        {
+            b->Arg((1 << i) * 1000);
+        }
+    }
+
+#define BM_REGISTER_TEMPLATE(FIXTURE, TESTNAME, ...) \
+        BENCHMARK_TEMPLATE_DEFINE_F(FIXTURE, TESTNAME, __VA_ARGS__)(benchmark::State& state) { Benchmark(state); } \
+        BENCHMARK_REGISTER_F(FIXTURE, TESTNAME)
+
+#define BM_REGISTER_SIZE_FIXTURES(FIXTURE, TESTNAME, ALLOCATORTYPE) \
+    BM_REGISTER_TEMPLATE(FIXTURE, TESTNAME##_SMALL, ALLOCATORTYPE, SMALL)->Apply(RunRanges); \
+    BM_REGISTER_TEMPLATE(FIXTURE, TESTNAME##_BIG, ALLOCATORTYPE, BIG)->Apply(RunRanges); \
+    BM_REGISTER_TEMPLATE(FIXTURE, TESTNAME##_MIXED, ALLOCATORTYPE, MIXED)->Apply(RunRanges);
+
```

Comment:
Testing threaded allocator performance might be interesting too ?
```cpp
ThreadRange(1, 4)->Apply(RunRanges);
```

Reply:
Great suggestion, will add